doc/manuals/sprokit/embedded_pipeline.rst

```diff
@@ -0,0 +1,327 @@
+Embedded Pipeline
+=================
+
+The embedded pipeline class provides a way to instantiate and run a
+sprokit pipeline within a program. The main use case for an embedded
+pipeline is in cases where a program calculates or obtains the input
+and needs the output for local processing. A prime example would be a
+GUI that runs a pipeline and display the results.
+
+The pipeline to be run is passed to the ``build_pipeline()`` method as a
+stream. The stream can be from a file or from a string built at run
+time. If the pipeline is coming from a file, it is useful to supply
+the directory portion (up to but not including the last '/') to the
+call so the error messages can provide the real file name and any
+*relativepath* modifiers can supply an accurate prefix. If the
+pipeline comes from a string stream, do not supply this parameter.
+
+After the pipeline has been built, it is started with the ``start()``
+call. Data are supplied to the pipeline via the ``send()`` method and
+retrieved with the ``receive()`` method. If input data is supplied
+faster than the pipeline can accept it, the ``send()`` method will
+block until there is room for the new element. Similarly, the
+``receive()`` method will block if there is no output available from the
+pipeline.
+
+When all the input has been supplied to the pipeline, the
+``send_end_if_input()`` is used to signal this condition. This will
+cause the pipeline terminate after all the supplied data has been
+processed. The ``receive()`` method will return an end of data
+indication after the last pipeline output has been returned.
+
+Example
+-------
+
+Lets look at an example to see the details of using an embedded
```

Comment:
"Let's" (note apostrophe)... although this is rather informal, and I wonder if this sentence is even necessary?